dinersclub/giftcards.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"encoding/json"
+	"time"
+	"context"
+	"fmt"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/vlab-research/go-reloadly/reloadly"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+type GiftCardsProvider struct {
+	pool *pgxpool.Pool
+	svc  *reloadly.Service
+}
+
+func NewGiftCardsProvider(pool *pgxpool.Pool) (Provider, error) {
+	cfg := getConfig()
+	svc := reloadly.NewGiftCards()
+	if cfg.Sandbox {
+		svc.Sandbox()
+	}
+	return &GiftCardsProvider{pool, svc}, nil
+}
+
+func (p *GiftCardsProvider) formatError(res *Result, err error, details *json.RawMessage) (*Result, error) {
```

Comment:
Let's abstract the shared bits between the Reloadly topups and Reloadly gift cards, instead of copy/pasting!

Reply:
Note that for GiftCards we return `INVALID_GIFT_CARD_DETAILS`, https://github.com/vlab-research/fly/pull/53/files#diff-c3d1b812fc10d0fbe337d8c7df8ae17b07ef69053e80080be2fd9f3d4c754f21R44. 

Whereas for TopUps we return `INVALID_PAYMENT_DETAILS`, https://github.com/vlab-research/fly/blob/3bb79a63280470cfca1f05de03556db87f031023/dinersclub/reloadly.go#L53

Alternatively, we could also agree on an error string that works for both, GiftCards and TopUps.